tests/meson.build

```diff
@@ -1,3 +1,16 @@
+
+# gtest conditionals
+#
+if get_option('use_gtest').disabled()
+  subdir_done()
+endif
```

Comment:
I didn't know about `subdir_done()` - cool.

It introduces a problem for future, still - it means all other kinds of tests must be declared above unit tests (and this might be not the best way to do organize this file). I also don't like that these two `subdir_done` make description in comment below ('gtest dependency') inaccurate.

I suggest the following change:

change code between lines 25 and 30 to:
```meson
if (get_option('buildtype') == 'release' and get_option('unit_tests').auto())
  gtest_dep = disabler()
else
  gtest_dep = dependency('gtest', main : true,
                         required : get_option('unit_tests'),
                         fallback : ['gtest', 'gtest_main_dep'])
endif
```
I think it works exactly as we need it to work.

(please note I skipped `message`s - Meson already generates good message on its own when written this way)

And then adjust the comment explaining gtest dependency to the new functionality.

Reply:
Replaced as-is, and tested. Works very nice! Thanks for this improvement @dreamer .
Also updated the commit comment.